sdk/eventhubs/azure-messaging-eventhubs/README.md

```diff
@@ -1,3 +1,189 @@
+Examples for various scenarios can be found on [pkg.go.dev](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs#pkg-examples) or in the example*_test.go files in our GitHub repo for [azeventhubs](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/messaging/azeventhubs).
+
+## Send events
+
+The following example shows how to send events to an event hub:
+
+```cpp
+// Your Event Hubs namespace connection string is available in the Azure portal.
+std::string connectionString = "<connection_string>";
+std::string eventHubName = "<event_hub_name>";
+
+Azure::Messaging::EventHubs::EventDataBatchOptions batchOptions;
+batchOptions.PartitionId = "1";
+Azure::Messaging::EventHubs::EventDataBatch eventBatch(batchOptions);
+
+Azure::Messaging::EventHubs::Models::EventData message;
+message.Body.Data = {'H', 'e', 'l', 'l', 'o', '2'};
+
+eventBatch.AddMessage(message);
+
+Azure::Messaging::EventHubs::ProducerClientOptions producerOptions;
+producerOptions.Name = "sender-link";
+producerOptions.ApplicationID = "some";
+
+auto client = Azure::Messaging::EventHubs::ProducerClient(
+      connectionString, eventHubName, producerOptions);
+auto result = client.SendEventDataBatch(eventBatch);
+```
+
+## Receive events
+
+The following example shows how to receive events from partition 1 on an event hub:
+
+```cpp
+// Your Event Hubs namespace connection string is available in the Azure portal.
+std::string connectionString = "<connection_string>";
+std::string eventHubName = "<event_hub_name>";
+
+Azure::Messaging::EventHubs::ConsumerClientOptions consumerOptions;
+consumerOptions.RetryOptions.Mode = Azure::Core::Retry::Mode::Fixed;
+consumerOptions.RetryOptions.MaxRetries = 1;
+
+auto client = Azure::Messaging::EventHubs::ConsumerClient(
+	connectionString, eventHubName, "$Default", consumerOptions);
+
+Azure::Messaging::EventHubs::PartitionClientOptions partitionOptions;
+// We want to consume all messages from the earliest (the default behavior is to consume new events).
+partitionOptions.StartPosition.Inclusive = true;
+partitionOptions.StartPosition.Earliest = true;
+Azure::Messaging::EventHubs::PartitionClient partitionClient
+        = client.CreatePartitionClient("1", partitionOptions);
+
+auto events = partitionClient.ReceiveEvents(1);
+```
+
+
+# Troubleshooting
+
+## Logging
+
+1. The EventHubs SDK client uses the [Azure SDK log message](https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/core/azure-core#sdk-log-messages) functionality to 
+enable diagnostics.
+1. In addition to the direct traces, the [ProducerClient][producer_client] and [ConsumerClient][consumer_client] classes also support logging of the events that are being sent/received.
+
+To enable additional logging, you can enable the `VerboseLogging` client option field:
+
+```cpp
+  Azure::Messaging::EventHubs::ProducerClientOptions producerOptions;
+  producerOptions.Name = "sender-link";
+  producerOptions.VerboseLogging = true;
```

Comment:
Why do we have a different way to tailor log level (as an option) rather than using the existing SDK-wide approach
https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/core/azure-core/README.md#sdk-log-messages
`Logger::SetLevel(Logger::Level::Verbose);`

Reply:
AMQP generates an obscene level of logging, far beyond what the SDK normally produces. At a level that makes it extremely difficult to figure out what's actually happening. 

The VerboseLogging flag enables all that logging. 

The AMQP logging is ALSO controlled by the existing SDK-wide approach.